app/controllers/admin/api/services/proxies_controller.rb

```diff
@@ -63,13 +63,7 @@ def show
   ##~ op.parameters.add name: "jwt_claim_with_client_id_type", description: "JWT Claim With ClientId Type. Either `plain` or `liquid`", dataType: "string", paramType: "query", required: false
   #
   def update
-    if proxy.update_attributes(proxy_params)
-      if proxy.service_mesh_integration?
-        proxy.deploy!
-      elsif proxy.apicast_configuration_driven
-        proxy.deploy_v2
-      end
-    end
+    proxy.deploy! if proxy.update_attributes(proxy_params) && !apiap?
```

Comment:
Why changing the logic here? Calling just `proxy.deploy!` is not the same as we had before. Now it will also check `Proxy#deployable?` and potentially call `Proxy#deploy_v1`.

Reply:
There is definitely some duplicated and odd behavior here, although I refactored it wrong.

For example when service mesh integration, the same condition is run 3 times:
```ruby
if proxy.service_mesh_integration?         # true (proxies_controller.rb)
  if ([...] || service_mesh_integration?)  # again, true (proxy.rb)
    if service_mesh_integration?           # 3rd time, true (proxy.rb)
      deploy_service_mesh_integration
```
In case of apicast, it bypasses `deployable?` when run from the controller, but it doesn't when run from the model. That's inconsistent, isn't it? Could you clarify this to me?

Finally for `deploy_v1` you're right.

Reply:
I'm leaving everything like it is here and proposing a simplification in #1327